trace-utils/src/trace_utils.rs

```diff
@@ -40,6 +45,210 @@ pub async fn get_traces_from_request_body(body: Body) -> anyhow::Result<(usize,
     Ok((size, traces))
 }
 
+#[inline]
+fn get_v5_strings_dict(reader: &mut Reader<impl Buf>) -> anyhow::Result<Vec<String>> {
+    let dict_size =
+        read_array_len(reader).map_err(|err| anyhow!("Error reading dict size: {err}"))?;
+    if dict_size as i32 > 25 * 1e6 as i32 {
+        anyhow::bail!(
+            "Error deserializing strings dictionary. Dict size is too large: {dict_size}"
+        );
+    }
+    let mut dict: Vec<String> = Vec::with_capacity(dict_size.try_into()?);
+    for _ in 0..dict_size {
+        let val: Value = read_value(reader)?;
+        match val {
+            Value::String(s) => {
+                let s = s.to_string().replace(&['\"'][..], "");
+                dict.push(s);
+            }
+            _ => anyhow::bail!("Error deserializing strings dictionary. Value in string dict is not a string: {val}")
+        }
+    }
+    Ok(dict)
+}
+
+#[inline]
+fn get_v5_span(reader: &mut Reader<impl Buf>, dict: &[String]) -> anyhow::Result<Span> {
+    let mut span: Span = Default::default();
+    let span_size = rmp::decode::read_array_len(reader)
+        .map_err(|err| anyhow!("Error reading span size: {err}"))? as usize;
+    if span_size != 12 {
+        anyhow::bail!("Expected an array of exactly 12 elements in a span, got {span_size}");
+    }
+    //0 - service
```

Comment:
So values come in order and I imagine that's why we are doing 12 reads here?

Reply:
Yup, msgpack is self describing but as far as I know all fields are required here. Maybe Bjorn can confirm?

Reply:
According to the docs, all are required: https://github.com/DataDog/datadog-agent/blob/579d60cbfc8635559338b19fa855bf59b6d10f00/pkg/trace/api/version.go#L110